tests/integration/conftest.py

```diff
@@ -0,0 +1,1064 @@
+        """The total number of test cases in this collection.
+
+        Returns:
+            int:  The total number of test cases in this collection.
+        """
+
+        return self._total_count
+
+    @property
+    def skip_count(self):
+        """The total number of test cases with 'skipped' state in this collection.
+
+        Returns:
+            int:  The total number of test cases with 'skipped' state in this collection.
+        """
+
+        return self._skip_count
+
+    @property
+    def failure_count(self):
+        """The total number of test cases with 'failed' state in this collection.
+
+        Returns:
+            int:  The total number of test cases with 'failed' state in this collection.
+        """
+
+        return self._failure_count
+
+    @property
+    def error_count(self):
+        """The total number of test cases with 'errored' state in this collection.
+
+        Returns:
+            int:  The total number of test cases with 'errored' state in this collection.
+        """
+
+        return self._error_count
+
+    @property
+    def total_duration(self):
+        """The total duration for all tests in this collection.
+
+        Returns:
+            int:  The total duration for all tests in this collection.
+        """
+
+        return self._total_duration
+
+    def add_test_case(self,
+                      state,
+                      name=None,
+                      class_name=None,
+                      file_path=None,
+                      line=None,
+                      start=None,
+                      duration=None,
+                      message=None,
+                      jira_tickets=None):
+        """Add a test case to the collection of test cases.
+
+        Args:
+            state (str): The state of the test. ('passed', 'skipped', 'failed', 'errored')
+            name (str): The desired name of the test. (Automatically generated if value is None)
+            class_name (str): The desired classname of the test. (Automatically generated if value is None)
+            file_path (str): The desired file path of the test. (Automatically generated if value is None)
+            line (int): The line number at which the test case can be found in the "file_path".
+                (Automatically generated if value is None)
+            start (datetime.datetime): Execution start time. (Automatically generated if value is None)
+            duration (int): The desired duration of the test in seconds. (Automatically generated if value is None)
+            message (str): The desired message for the test which is only used for 'skipped', 'failed', 'errored'
+                test states. (Automatically generated if value is None)
+            jira_tickets (list(str)): A list of Jira ticket IDs to associated with the test case.
+                (Automatically generated if value is None)
+
+        Raises:
+            RuntimeError: Invalid value provided for the 'state' argument.
```

Comment:
This statement appears to be invalid since an invalid value for state is not trapped in this method.